agent/lib/kontena/workers/container_log_worker.rb

```diff
@@ -57,12 +58,24 @@ def on_message(id, stream, chunk)
       return unless match
       time = DateTime.parse(match[1])
       data = match[2]
-      @queue << {
-          id: id,
-          time: time.utc.xmlschema,
-          type: stream,
-          data: data
+      msg = {
+          event: 'container:log',
+          data: {
+              id: id,
+              service: @container.service_name,
+              stack: @container.stack_name,
+              instance: @container.instance_number,
+              time: time.utc.xmlschema,
+              type: stream,
+              data: data
+          }
       }
+      @queue << msg
+      publish_log(msg.dig('data', 'data'))
```

Comment:
Yes, the `FluentdWorker` expects to get the service related data to be able to tag the event properly.